teps/0080-support-domainscoped-parameterresult-names.md

```diff
@@ -0,0 +1,165 @@
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Allow Task and Pipeline authors to expose parameters and results that allow
+`.` characters.
+
+> _This TEP was originally raised [here](https://github.com/tektoncd/pipeline/issues/3590)
+as an issue for discussion._
+
+## Motivation
+
+The motivation of this work is to *enabl* us to establish conventions around the
+definition of parameters and results that may have a deeper meaning for
+higher-level systems without a high-degree of accidental naming collisions.
+
+The pursuit of `.` is specifically to follow the precedents set by many other systems
+(including Kubernetes labels, CloudEvent types, Java import paths, ContainerD plugins,
+and many more) of using a Domain-scoped naming convention to "claim" a segment of
+names for the owner's exclusive use.  This is of course conventional, but in practice
+is generally good enough.
+
+With this work, higher-level systems could start to define Task/Pipeline "interfaces"
+or leverage partial signature matching to enable special semantics around certain
+signatures, without (significant) concerns around matching something accidentally.
+
+> _It is notable that with @bobcatfish [TEP for stronger
+typing](https://github.com/tektoncd/community/pull/479) that these conventions should
+also establish the types associated with those parameter names, and might form a good
+case for SchemaRefs._
+
+### Goals
+
+Enable conventions to be established around the use of domain-scoped names as a way
+for the domain owner to "own" the definition of what those parameters and results are
+for and how they will be used.
+
+### Non-Goals
+
+It is not a goal of this TEP to establish these conventions, or begin to define any
+"well known" parameters or results owned by Tekton (e.g. `dev.tekton.foo.bar`).
+
+
+### Use Cases (optional)
+
+See the [original issue](https://github.com/tektoncd/pipeline/issues/3590) for some
+of these.
+
+There are potentially a lot more things like this, but I'd rather leaves those for
+a conversation around the types of conventions we should standardize, and not rat hole
+here (just about enabling that next convo).
+
+## Requirements
+
+Parameters and Result name resolution must be unamiguous, especially in the presence
+of proposals like [TEP for stronger typing](https://github.com/tektoncd/community/pull/479),
+which allows folks to access fields of complex parameters.
+
+## Proposal
+
+Two parts to the proposal:
+
+1. Allow folks to define parameters and results with `.` in the name:
+```yaml
+  params:
+    - name: dev.mattmoor.foo
```

Comment:
my biggest concern around allowing this is that it would conflict with expanding json path support - you mentioned that in https://github.com/tektoncd/community/pull/479#discussion_r692364657 but id like to understand how these would work together

(maybe we'd have to always use some escaping in order to support the dots?)

Reply:
See bullet 2 right below this?  😄

Reply:
im wondering if the json path related proposals would make you want to change the syntax at all - i dont think this proposal mentions json path syntax at all so im wondering if `"dev.mattmoor.foo"` is the syntax we want to go with if we're embracing more jsonpath support

maybe all we need to make this happen is json path `[]` syntax? e.g. `$(params.[dev.mattmoor.foo])` (or maybe json supports some kind of additional escaping syntax like `params.dev\.mattmoor\.foo`?

Reply:
Yeah, I'm happy to change it.  Will do it when I do another pass over the markdown itself.

Reply:
Ok, updated.  LMK if this is consistent with your expectations, and if not I'm happy to adjust!

Reply:
thanks @mattmoor !!